src/modules/land_detector/MulticopterLandDetector.cpp

```diff
@@ -180,7 +182,17 @@ bool MulticopterLandDetector::_get_ground_contact_state()
 	    (!verticalMovement || !_has_position_lock())) {
 		return true;
 	}
+	// Check if we are moving vertically - here only check in Position control mode or Alt mode
+	if(_manual.z >= _params.hoverThrottle *0.8f )
+		return false;
+	//if _vehicleLocalPosition.vz  <   50% * ( Command velocity )  ,there are 50% margin to command velocity ,but need vertical velocity much accurately
+	verticalMovement = fabsf(_vehicleLocalPosition.vz) > (_params.hoverThrottle-_manual.z)*_params.maxDownRate * armThresholdFactor;
```

Comment:
we talk this in  _control_mode.flag_control_climb_rate_enabled and manual mode ，thanks to not good gps,we land not good. 
_manual.z from 0 to 0.5 means vertical velocity from - MPC_Z_VEL_MAX m/s to 0 m/s   because of dead zone，so I just check (_manual.z >= _params.hoverThrottle *0.8f) .  this means at lest we have 0.1m/s error when throttle about 40% when MPC_Z_VEL_MAX =1m/s , because 40% throttle  means 0.2m/s down speed, and more margin  when throttle   lower. there about 1s GROUND_CONTACT_TRIGGER_TIME_US =1000000 ，maybe smaller if safe